torch/nn/utils/_expanded_weights/linear_expanded_weights.py

```diff
@@ -0,0 +1,23 @@
+import torch
+import torch.nn.functional as F
+from .expanded_weights_impl import implements_per_sample_grads
+from .expanded_weights_utils import forward_helper, grad_if_exists, grad_if_exists_for_input, unpack_expanded_weight_or_tensor
+
+@implements_per_sample_grads(F.linear)
+class LinearPerSampleGrad(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, *expanded_args):
+        if len(expanded_args[0].shape) <= 1:
+            raise RuntimeError("Input does not have a batch dimension. Expanded Weights "
+                               f"expected input of at least rank 2, got of rank {len(expanded_args[0].shape)}")
+        return forward_helper(F.linear, ctx, expanded_args, 1)
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        (input, weight, bias) = ctx.args
+        results = []
+
+        results.append(grad_if_exists_for_input(input, lambda: grad_output.matmul(unpack_expanded_weight_or_tensor(weight))))
```

Comment:
Hmm, we're really baking in the assumption that the ExpandedWeight is wrapping a leaf tensor (a tensor that requires grad that is a leaf of the computation graph). We probably want a more generalized version of ExpandedWeight in the future, where users can e.g. call .relu() on an ExpandedWeight object to return a new ExpandedWeight object.

How much do you think we should worry about what we need in the future? I'm not sure -- I think Opacus bakes in this assumption as well? If we wanted to test our hypothesis that "this assumption works for most models" then we should do something like demonstrate running ExpandedWeights over the torchbenchmark suite of models. OTOH we're not exposing this yet, so another approach would be to merge this first "as a prototype" and then validate next.